scripts/utils/htmlToMarkdown.js

```diff
@@ -0,0 +1,124 @@
+const Turndown = require('turndown');
+const HTMLtoJSX = require('htmltojsx');
+const { extractTags } = require('./node');
+
+const escapes = [
+  [/\\/g, '\\\\'],
+  [/\*/g, '\\*'],
+  [/^-/g, '\\-'],
+  [/^\+ /g, '\\+ '],
+  [/^(=+)/g, '\\$1'],
+  [/^(#{1,6}) /g, '\\$1 '],
+  [/`/g, '\\`'],
+  [/^~~~/g, '\\~~~'],
+  [/\[/g, '\\['],
+  [/\]/g, '\\]'],
+  [/_/g, '\\_'],
+  [/^(\d+)\. /g, '$1\\. '],
+
+  // Because we are converting to JSX, we need to ensure opening and closing < >
+  // are properly escaped to their HTML entities, otherwise it is parsed as the
+  // start of a component.
+  [/</g, '&lt;'],
+  [/>/g, '&gt;'],
+];
+
+// We need to override the built-in `escape` function so that we can use our
+// updated `escapes` variable from above, which differs slightly than the
+// default `escapes` that ships with turndown.
+Turndown.prototype.escape = (string) => {
+  return escapes.reduce((accumulator, escape) => {
+    return accumulator.replace(escape[0], escape[1]);
+  }, string);
+};
+
+const turndown = new Turndown({
+  headingStyle: 'atx',
+  codeBlockStyle: 'fenced',
+  fence: '```',
+});
+
+const htmlToJSXConverter = new HTMLtoJSX({ createClass: false });
+
+// TODO: Use with components we want to keep
+// const replaceWithContent = (node, content) => {
+//   const openingTag = node.outerHTML.slice(
+//     0,
+//     node.outerHTML.indexOf(node.innerHTML)
+//   );
+//   const closingTag = `</${node.tagName.toLowerCase()}>\n`;
+
+//   const outerJSX = htmlToJSXConverter.convert(`${openingTag}|||${closingTag}`);
+
+//   return outerJSX.replace('|||', `\n${content}\n`).trim();
+// };
+
+const repeat = (character, count) => Array(count + 1).join(character);
+
+turndown
+  .addRule('codeBlocks', {
+    filter: ['pre'],
+    replacement: (_content, node, options) => {
+      const code = node.textContent;
+      const fenceChar = options.fence.charAt(0);
+      const fenceInCodeRegex = new RegExp(`^${fenceChar}{3,}`, 'gm');
+
+      let fenceSize = 3;
+      let match;
+
+      while ((match = fenceInCodeRegex.exec(code))) {
```

Comment:
could you explain what's going on here? I'm a bit confused

Reply:
I meant to go back and add a comment here that points to the original source (which I will do as a followup). This comes from the [original implementation](https://github.com/domchristie/turndown/blob/master/src/commonmark-rules.js#L111-L134) in Turndown. This handles the case of a nested fenced block inside of another. For example, our [developer website component guide](https://github.com/newrelic/developer-website/blob/develop/COMPONENT_GUIDE.md#usage-4) has examples of using fenced code blocks:

````
```jsx
console.log('Hello')
```
````

Note here how I had to wrap that with 4 backticks to get it to render the 3 backticks in the code block. This `while` essentially handles that edge case.

Does that make sense?

Reply:
ahhh yes that makes perfect sense